libs/ledger-live-common/src/bot/portfolio/process-sync.ts

```diff
@@ -0,0 +1,318 @@
+  const [family, key] = process.argv.slice(2);
+  const spec: AppSpec<any> = allSpecs[family][key];
+
+  const { COINAPPS, SEED } = process.env;
+
+  if (!COINAPPS) {
+    throw new Error("COINAPPS env variable is required");
+  }
+  if (!SEED) {
+    throw new Error("SEED env variable is required");
+  }
+
+  // Prepare speculos device simulator
+
+  const appCandidates = await listAppCandidates(COINAPPS);
+
+  const { appQuery, currency, dependency } = spec;
+  const appCandidate = findAppCandidate(appCandidates, appQuery);
+  if (!appCandidate) {
+    console.warn("no app found for " + spec.name);
+    console.warn(appQuery);
+    console.warn(JSON.stringify(appCandidates, undefined, 2));
+  }
+  if (!appCandidate) {
+    throw new Error(
+      `no app found for ${spec.name}. Are you sure your COINAPPS is up to date?`
+    );
+  }
+  const deviceParams = {
+    ...(appCandidate as AppCandidate),
+    appName: spec.currency.managerAppName,
+    seed: SEED,
+    dependency,
+    coinapps: COINAPPS,
+  };
+
+  const device = await createSpeculosDevice(deviceParams);
+
+  try {
+    const audit = new Audit(bootTime);
+
+    // We scan and synchronize the accounts
+
+    const localCache = {};
+    const cache = makeBridgeCacheSystem({
+      saveData(c, d) {
+        localCache[c.id] = d;
+        return Promise.resolve();
+      },
+
+      getData(c) {
+        return Promise.resolve(localCache[c.id]);
+      },
+    });
+
+    const bridge = getCurrencyBridge(currency);
+    const syncConfig = {
+      paginationConfig: {},
+    };
+
+    await cache.prepareCurrency(currency);
+    const accounts = await bridge
+      .scanAccounts({
+        currency,
+        deviceId: device.id,
+        syncConfig,
+      })
+      .pipe(
+        filter((e) => e.type === "discovered"),
+        map((e) => e.account),
+        reduce<Account, Account[]>((all, a) => all.concat(a), []),
+        timeoutWith(
+          getEnv("BOT_TIMEOUT_SCAN_ACCOUNTS"),
+          throwError(
+            new Error("scan accounts timeout for currency " + currency.name)
+          )
+        )
+      )
+      .toPromise();
+
+    audit.end();
+
+    const accountsRaw = JSON.stringify(accounts.map(toAccountRaw));
+    const preloadJSON = JSON.stringify(localCache);
+    audit.setAccountsJSONSize(accountsRaw.length);
+    audit.setPreloadJSONSize(preloadJSON.length);
+
+    /*
+    // TODO big data
+    if (REPORT_FOLDER) {
+      const accountsFile = path.join(REPORT_FOLDER, "accounts.json");
+      await fs.writeFile(accountsFile, accountsRaw);
+    }
+    */
+
+    report.refillAddress = accounts[0]?.freshAddress;
+    report.accountBalances = accounts.map((a) => a.balance.toString());
+    report.accountIds = accounts.map((a) => a.id);
+    report.accountOperationsLength = accounts.map((a) => a.operations.length);
+    report.auditResult = audit.result();
+  } finally {
+    await releaseSpeculosDevice(device.id);
+  }
+
+  return report;
+}
+
+class SlowFrameDetector {
```

Comment:
Would it make sense to separate classes into their one files ? I feel like you did a great job at separating the logic already